### PR TITLE
[Feature] : Add ScatterND TensorRT Plugin

### PR DESCRIPTION
**Pull request by grimoire**
This PR add support to ONNX [ScatterND](https://github.com/onnx/onnx/blob/master/docs/Operators.md#ScatterND)
Support convert of pytorch code such as:
```python
tensor[:, -2:] += 1
```

### CODE REVIEW

File: mmcv/ops/csrc/tensorrt/plugins/trt_scatternd.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      case 1:
+        // indices
+        return inOut[pos].type == nvinfer1::DataType::kINT32 &&
+               inOut[pos].format == nvinfer1::TensorFormat::kLINEAR;
+      case 2:
+        // updates
+        return inOut[pos].type == inOut[0].type &&
+               inOut[pos].format == inOut[0].format;
+      default:
+        return true;
```
Comment (RunningLeon) [inline mmcv/ops/csrc/tensorrt/plugins/trt_scatternd.cpp:69]: Will there have any problem if `return true` is set by default?

File: mmcv/ops/csrc/tensorrt/plugins/trt_scatternd.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
+#include "trt_scatternd.hpp"
```
Comment (RunningLeon) [inline mmcv/ops/csrc/tensorrt/plugins/trt_scatternd.cpp:1]: Would add unit test for  `scatternd plugin`.

File: mmcv/ops/csrc/tensorrt/plugins/trt_scatternd.cpp (hunk 1 above)
Comment (grimoire) [reply mmcv/ops/csrc/tensorrt/plugins/trt_scatternd.cpp:69]: Theoretically, switch() include all possible case. Default branch will never be reached.

File: mmcv/ops/csrc/tensorrt/plugins/trt_scatternd_kernel.cu [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+using mmcv::TensorDesc;
+
+template <typename T>
+__global__ void onnx_scatternd_kernel(const int n, const int* indices,
+                                      const T* update, T* output,
+                                      TensorDesc tensor_desc,
+                                      TensorDesc indice_desc) {
+  const int indice_cols = indice_desc.shape[indice_desc.dim - 1];
+  const int copy_stride = tensor_desc.stride[indice_cols - 1];
+  const int* shape = &(tensor_desc.shape[0]);
```
Comment (RunningLeon) [inline mmcv/ops/csrc/tensorrt/plugins/trt_scatternd_kernel.cu:20]: `shape` is unused. Could be removed.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by hellock — APPROVED** [review]

**Pull request merged**